src/scripting/mapgen_lua_kernel.cpp

```diff
@@ -157,34 +157,55 @@ static int intf_default_generate_height_map(lua_State *L)
 }
 /**
  * Finds a path between two locations.
- * - Args 1,2: source location.
- * - Args 3,4: destination.
- * - Arg 5: cost function
- * - Args 6,7 size of map.
- * - Arg 8 include border.
+ * - Args 1: source location.
+ * - Args 2: destination.
```

Comment:
This is meant to still work as the deprecated `wesnoth.find_path`, there's no wrapper function around it in `data/lua/core/_initial.lua`.

Reply:
I'm not sure what you're getting at? There's an assignment for `wesnoth.find_path = wesnoth.paths.find_path` on line 112. The arguments to the new function are the same but with two extra features - the first argument can be a location table (rather than the first two arguments being x and y) and the last argument can be an options table instead of passing the cost function and other options all as separate parameters.

So, any code written against the current API still works.

Reply:
> the first argument can be a location table

Then the documentation doesn't match the function.

Reply:
It does match. The first argument is a location. The computer might see it as one argument (a table with two values) or as two arguments (separate integers), but it's logically a single argument.

And this equivalence holds true basically everywhere you can pass a location in the Lua API; all I did was add another case of it.